docs/index.bs

```diff
@@ -2181,62 +2157,64 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
+            1. Resolve |promise| with false.
+        1. Return |promise|.
     </section>
 
     <section algorithm="cache-storage-open">
       <h4 id="cache-storage-open">{{CacheStorage/open(cacheName)}}</h4>
 
       <dfn method for="CacheStorage"><code>open(|cacheName|)</code></dfn> method *must* run these steps:
 
-        1. Let |p| be a new <a>promise</a>.
-        1. Run the following substeps:
-            1. [=map/For each=] |key| → |value| of <a>name to cache map</a>:
+        1. Let |promise| be [=a new promise=].
+        1. Run the following substeps [=in parallel=]:
+            1. [=map/For each=] |key| → |value| of the [=name to cache map=]:
                 1. If |cacheName| matches |key|, then:
-                    1. Resolve |p| with a new {{Cache}} object which is a copy of |value|.
+                    1. Resolve |promise| with a new {{Cache}} object that represents |value|.
                     1. Abort these steps.
-            1. Let |cache| be a new {{Cache}} object.
-            1. [=map/Set=] <a>name to cache map</a>[|cacheName|] to |cache|. If this cache write operation failed due to exceeding the granted quota limit, reject |p| with a "{{QuotaExceededError}}" {{DOMException}} and abort these steps.
-            1. Resolve |p| with |cache|.
-        1. Return |p|.
+            1. Let |cache| be a new [=request response list=].
+            1. [=map/Set=] the [=name to cache map=][|cacheName|] to |cache|. If this cache write operation failed due to exceeding the granted quota limit, reject |promise| with a "{{QuotaExceededError}}" {{DOMException}} and abort these steps.
+            1. Resolve |promise| with a new {{Cache}} object that represents |cache|.
+        1. Return |promise|.
     </section>
 
     <section algorithm="cache-storage-delete">
       <h4 id="cache-storage-delete">{{CacheStorage/delete(cacheName)}}</h4>
 
       <dfn method for="CacheStorage"><code>delete(|cacheName|)</code></dfn> method *must* run these steps:
 
-        1. Let |p| be the result of running the algorithm specified in {{CacheStorage/has(cacheName)}} method with |cacheName| as the argument.
-        1. Return the result of <a>transforming</a> |p| with a fulfillment handler that, when called with argument |cacheExists|, performs the following substeps <a>in parallel</a>:
+        1. Let |promise| be the result of running the algorithm specified in {{CacheStorage/has(cacheName)}} method with |cacheName|.
+        1. Return the result of [=transforming=] |promise| with a fulfillment handler that, when called with argument |cacheExists|, performs the following substeps [=in parallel=]:
             1. If |cacheExists| is true, then:
-                1. Delete a <a>Record</a> {\[[key]], \[[value]]} <var ignore>entry</var> from its <a>name to cache map</a> where |cacheName| matches entry.\[[key]].
+                1. [=map/Remove=] the [=name to cache map=][|cacheName|].
                 1. Return true.
```

Comment:
I'm not sure we can "return" from in parallel steps.

Reply:
I'm not sure we can. For this particular case, I removed "in parallel". I think that's okay as the fulfillment handler's already scheduled async in the microtask queue. But for other similar cases, I didn't change them in this PR as I wasn't sure if they can all run in the main thread. I'll look at them as a separate work.

Reply:
Although it's on the microtask queue, it's still blocking the event loop. I think we need to create a promise and return it.

Reply:
Yes, actually the fulfillment handler steps were very much incorrect. I made them run in the event loop and create a promise there and resolve/reject the promise from the parallel steps. Also, while changing them, I changed the interface of Batch Cache Operations such that it runs synchronously without returning a promise and the call sites invoke it from a promise job.